common/wireaddr.c

```diff
@@ -1,15 +1,70 @@
 #include <arpa/inet.h>
 #include <assert.h>
+#include <ccan/io/io.h>
+#include <ccan/str/hex/hex.h>
 #include <ccan/tal/str/str.h>
 #include <common/type_to_string.h>
 #include <common/utils.h>
 #include <common/wireaddr.h>
+#include <errno.h>
+#include <fcntl.h>
 #include <netdb.h>
 #include <netinet/in.h>
 #include <sys/socket.h>
 #include <sys/types.h>
+#include <unistd.h>
 #include <wire/wire.h>
 
+
+#define BASE32DATA "abcdefghijklmnopqrstuvwxyz234567"
+
+
+char *b32_encode(char *dst, u8 *src, u8 ver) {
```

Comment:
Please specify in more depth the requirements on `dst`.  What size should it be and so on.  A good comment description here would be good.